### ISSUE TITLE
[resotocore][feat] Deferred directed edge types that have `id` as source/dest and search criteria as the other endpoint

### ISSUE THREAD

**Issue by lloesche** [someengineering/resoto#693]
### Motivation

If a collector wants to create edges between multiple accounts but those accounts are being collected on different workers there's currently no way to create those edges. Collectors already support a feature called deferred_edges where they can say "at some point in the future this resource should have an edge to another resource with the following search criteria" and upon graph merge these deferred edges are resolved. However this only works within the same `resotoworker` instance.

### Desired Behavior

It would be nice if there was a way to define edges like e.g. :
```
{"from": "BuQ8K5Xy03B_LG-ENNWBUw", "to": {"/ancestors.cloud.reported.id": "aws", "kind": "aws_ec2_instance", "id": "i-0e81c3bec031034b8"}}
```
Or the other way around (i.e. `from` being the search criteria and `to` the fixed id). This should allow 1:n or n:1 relationships.

**Comment by lloesche** [comment]
Related to #628

**Comment by aquamatthias** [comment]
Fixed via #916
